### PR TITLE
HistoricalRecords: add additional fields (ip and browser)

### PR DESCRIPTION
**Pull request by stsewd**

### CODE REVIEW

File: readthedocs/core/signals.py [hunk 1]
```diff
@@ -122,10 +124,18 @@ def delete_projects(sender, instance, *args, **kwargs):
...
+    if not history_instance:
+        return
+
     history_user = kwargs['history_user']
-    if history_instance and history_user:
+    if history_user:
         history_instance.extra_history_user_id = history_user.id
         history_instance.extra_history_user_username = history_user.username
 
+    request = getattr(HistoricalRecords.context, 'request', None)
```
Comment (humitos) [inline readthedocs/core/signals.py:135]: I think I don't understand how this works. Why `HistoricalRecords.context`, which seems to be something static, will have a dynamic `request` attribute? 🤔

Comment (humitos) [reply readthedocs/core/signals.py:135]: I found this example, and it seems that it's injected in the middleware. Super weird to me.

https://django-simple-history.readthedocs.io/en/latest/historical_model.html?highlight=context

Also, it seems that it should be always there and there is no need for `getattr()` and `if request`? 😕

**Review by humitos — COMMENTED** [review]
Looks good to me. Shouldn't we add a test to confirm that we are saving this data correctly?

File: readthedocs/core/signals.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/core/signals.py:135]: Yeah, it's weird how they pass the request, but that's on their docs. You can end without a request object when you save the object without hitting the middleware, this is if you save the object outside a request like a task or from the django shell.

Comment (humitos) [reply readthedocs/core/signals.py:135]: Good point! Thanks.

**Review by humitos — APPROVED** [review]

**Comment by stsewd** [comment]
Changed from GenericIPField to a plain Charfield to avoid the same problems we had with using an ip field on the audit models.

**Pull request merged**